builder/softlayer/step_capture_image.go

```diff
@@ -4,35 +4,81 @@ import (
 	"fmt"
 	"github.com/mitchellh/multistep"
 	"github.com/mitchellh/packer/packer"
+	"strings"
 )
 
 type stepCaptureImage struct{}
 
+func (self *stepCaptureImage) getBlockDeviceIds(blockDevices []interface{}) ([]int64) {
+	blockDeviceIds := make([]int64, len(blockDevices))
+	deviceCount := 0
+
+	for _, val := range blockDevices {
+		blockDevice := val.(map[string]interface{})
+		diskImage := blockDevice["diskImage"].(map[string]interface{})
+		name := diskImage["name"].(string)
+		id := int64(blockDevice["id"].(float64))
+
+		if !strings.Contains(name, "SWAP") {
+			blockDeviceIds[deviceCount] = id
+			deviceCount++
+		}
+	}
+
+	return blockDeviceIds[:deviceCount]
+}
+
 func (self *stepCaptureImage) Run(state multistep.StateBag) multistep.StepAction {
 	client := state.Get("client").(*SoftlayerClient)
 	ui := state.Get("ui").(packer.Ui)
-
-	ui.Say("Preparing for capturing the instance image. Image snapshot type is Flex.")
-
 	instance := state.Get("instance_data").(map[string]interface{})
-	instanceId := instance["globalIdentifier"].(string)
 	config := state.Get("config").(config)
+	instanceId := instance["globalIdentifier"].(string)
+	var imageId string
 
-	data, err := client.captureImage(instanceId, config.ImageName, config.ImageDescription)
-	if err != nil {
-		err := fmt.Errorf("Error while trying to capture an image from instance (id=%s). Error: %s", instanceId, err)
-		ui.Error(err.Error())
-		state.Put("error", err)
-		return multistep.ActionHalt
-	}
+	ui.Say(fmt.Sprintf("Preparing for capturing the instance image. Image snapshot type is '%s'.", config.ImageType))
+
+	if config.ImageType == "standard" {
```

Comment:
Can we use a constants for the different image types? I think it is cleaner than passing/comparing strings.